nodes/nomos-node/src/api/backend.rs

```diff
@@ -0,0 +1,295 @@
+{
+    type Error = hyper::Error;
+    type Settings = AxumBackendSettings;
+
+    async fn new(settings: Self::Settings) -> Result<Self, Self::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Self {
+            settings,
+            _attestation: core::marker::PhantomData,
+            _blob: core::marker::PhantomData,
+            _certificate: core::marker::PhantomData,
+            _membership: core::marker::PhantomData,
+            _vid: core::marker::PhantomData,
+            _verifier_backend: core::marker::PhantomData,
+            _tx: core::marker::PhantomData,
+            _storage_serde: core::marker::PhantomData,
+            _sampling_backend: core::marker::PhantomData,
+            _sampling_network_adapter: core::marker::PhantomData,
+            _sampling_rng: core::marker::PhantomData,
+            _sampling_storage: core::marker::PhantomData,
+        })
+    }
+
+    async fn serve(self, handle: OverwatchHandle) -> Result<(), Self::Error> {
+        let mut builder = CorsLayer::new();
+        if self.settings.cors_origins.is_empty() {
+            builder = builder.allow_origin(Any);
+        }
+
+        for origin in &self.settings.cors_origins {
+            builder = builder.allow_origin(
+                origin
+                    .as_str()
+                    .parse::<HeaderValue>()
+                    .expect("fail to parse origin"),
+            );
+        }
+
+        let app = Router::new()
+            .layer(
+                builder
+                    .allow_headers([CONTENT_TYPE, USER_AGENT])
+                    .allow_methods(Any),
+            )
+            .layer(TraceLayer::new_for_http())
+            .merge(SwaggerUi::new("/swagger-ui").url("/api-docs/openapi.json", ApiDoc::openapi()))
+            .route("/cl/metrics", routing::get(cl_metrics::<T>))
+            .route("/cl/status", routing::post(cl_status::<T>))
+            .route(
+                "/cryptarchia/info",
+                routing::get(
+                    cryptarchia_info::<
+                        T,
+                        S,
+                        SamplingBackend,
+                        SamplingNetworkAdapter,
+                        SamplingRng,
+                        SamplingStorage,
+                        SIZE,
+                    >,
+                ),
+            )
+            .route(
+                "/cryptarchia/headers",
+                routing::get(
+                    cryptarchia_headers::<
+                        T,
+                        S,
+                        SamplingBackend,
+                        SamplingNetworkAdapter,
+                        SamplingRng,
+                        SamplingStorage,
+                        SIZE,
+                    >,
+                ),
+            )
+            .route("/da/add_blob", routing::post(add_blob::<A, B, M, VB, S>))
+            .route(
+                "/da/get_range",
```

Comment:
Yeah, nice changes. Better to compose here than the over-abstraction of converting the endopoints in types. It was my fault though, have been thinking about this for time. So, nice catch.